rero_ils/modules/sru/views.py

```
@@ -33,14 +61,75 @@
+def _generate_etag(content):
+    """Generate an ETag hash from response content.
+
+    Args:
+        content: The response body as string or bytes.
+
+    Returns:
+        A quoted ETag string suitable for HTTP headers.
+    """
+    if isinstance(content, str):
+        content = content.encode("utf-8")
+    return f'"{hashlib.md5(content).hexdigest()}"'
```

Ruff (0.15.1)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)